htdocs/core/class/html.form.class.php

```diff
@@ -6468,7 +6474,7 @@ public function load_tva($htmlname = 'tauxtva', $selectedrate = '', $societe_ven
 		}
 
 		// Now we get list
-		$num = $this->load_cache_vatrates($code_country); // If no vat defined, return -1 with message into this->error
+		$num = $this->load_cache_vatrates($code_country, $type_vat); // If no vat defined, return -1 with message into this->error
```

Comment:
This load the database into a cache array. We should not filter on type here. Filter should be done later after lin 6532.
If you filter here, when you call the load_tva a second time on same page with a different value for type, you will get result of first call.